vault/plugin_catalog.go

```diff
@@ -140,6 +143,37 @@ func (p *pluginClient) Conn() grpc.ClientConnInterface {
 	return p.clientConn
 }
 
+func (p *pluginClient) Reload() error {
+	p.logger.Debug("reload external plugin process")
+	return p.reloadFunc()
+}
+
+// reloadExternalPlugin
+// This should be called with the write lock held.
+func (c *PluginCatalog) reloadExternalPlugin(name, id string) error {
+	extPlugin, ok := c.externalPlugins[name]
+	if !ok {
+		return fmt.Errorf("plugin client not found")
+	}
+	if !extPlugin.multiplexingSupport {
+		err := c.cleanupExternalPlugin(name, id)
+		if err != nil {
+			return err
+		}
+	}
+
+	pc, ok := extPlugin.connections[id]
```

Comment:
If there's no multiplexing support, why doesn't this always error? Is it because the `id` will always be `"single"` in that case?

Reply:
The "single" ID is only used on the grpc server. For the plugin catalog we assign an ID in the multiplexed or non-multiplexed case: https://github.com/hashicorp/vault/blob/94410157d94f113099ca4eaed2b9b764f9ac5988/vault/plugin_catalog.go#L234-L246